### PR TITLE
Fix Broken Links

### PR DESCRIPTION
**Pull request by geographika**
CI builds are currently failing due to broken links in the docs.

This pull request:

- updates any permanent URL redirects to the new URL
- fixes some of the broken links
- ignores checks to StackOverlow - the rate limiting causes these to fail, and makes the CI run for several minutes retrying

Questions:

- Should `doc/quickstart/vmware_quickstart.rst` be removed? It is not in the HTML output and is written for OSGeoLive 5.5.

### CODE REVIEW

**Comment by geographika** [comment]
The linkchecking is quite flaky. I think all URLs across all languages are checked on every pull request by https://github.com/OSGeo/OSGeoLive-doc/blob/master/.github/workflows/linkcheck.yml as the https://github.com/OSGeo/OSGeoLive-doc/blob/master/.github/workflows/locale-update.yml action updates the language files automatically. 

Maybe the linkcheck action should only be triggered by specific tags in a commit? Otherwise most doc pull requests will fail for unrelated reasons.

**Comment by geographika** [comment]
With the various fixes and increased retries, the checks currently seem stable (green ticks across all CI checks for 3 pushes in a row). Ready to merge from my side.

File: doc/conf.py [hunk 1]
```diff
@@ -227,19 +232,27 @@
...
-        # link exists but link check failes
+        # link exists but link check fails
         # SSL: CERTIFICATE_VERIFY_FAILED
         # Used in sponsors.rst
         'https://www.ntua.gr/en',
         'https://www.ice.ucdavis.edu/',
 
-        # Link with certiciate problems
+        # link exists, with strange redirects, but checks fail
+        'https://webchat.freenode.net/',
```
Comment (neteler) [inline doc/conf.py:242]:
```suggestion
        'https://web.libera.chat/',
```

AFAIK most OSGeo projects have abandoned freenode in favor of libera.

File: doc/conf.py [hunk 2]
```diff
@@ -227,19 +232,27 @@
...
+
+        # Link with certificate problems
         # Used in quickstart/liblas_quickstart
         #'https://epsg.org/home.html',
         'https://www.epsg.org/',
```
Comment (neteler) [inline doc/conf.py:247]:
```suggestion
        'https://epsg.org/',
```

(the www variant is broken)

File: doc/overview/ossim_overview.rst [hunk 3]
```diff
@@ -43,15 +43,15 @@ Details
 
 **Website:** http://www.ossim.org
```
Comment (neteler) [inline doc/overview/ossim_overview.rst:44]:
```suggestion
**Website:** https://trac.osgeo.org/ossim/
```

(ossim.org seems to be gone for years according to web.archive.org)

File: doc/quickstart/actinia_quickstart.rst [hunk 4]
```diff
@@ -473,4 +473,4 @@ What next?
 
 * Visit the actinia website at `https://actinia.mundialis.de <https://actinia.mundialis.de>`_
 * actinia tutorial: `https://neteler.gitlab.io/actinia-introduction <https://neteler.gitlab.io/actinia-introduction>`_
-* Further reading: Neteler, M., Gebbert, S., Tawalika, C., Bettge, A., Benelcadi, H., Löw, F., Adams, T, Paulsen, H. (2019). Actinia: cloud based geoprocessing. In Proc. of the 2019 conference on Big Data from Space (BiDS'2019) (pp. 41-44). EUR 29660 EN, Publications Office of the European Union 5, Luxembourg: P. Soille, S. Loekken, and S. Albani (Eds.). (`DOI <https://zenodo.org/record/2631917>`__)
+* Further reading: Neteler, M., Gebbert, S., Tawalika, C., Bettge, A., Benelcadi, H., Löw, F., Adams, T, Paulsen, H. (2019). Actinia: cloud based geoprocessing. In Proc. of the 2019 conference on Big Data from Space (BiDS'2019) (pp. 41-44). EUR 29660 EN, Publications Office of the European Union 5, Luxembourg: P. Soille, S. Loekken, and S. Albani (Eds.). (`DOI <https://zenodo.org/records/2631917>`__)
```
Comment (neteler) [inline doc/quickstart/actinia_quickstart.rst:476]:
```suggestion
* Further reading: Neteler, M., Gebbert, S., Tawalika, C., Bettge, A., Benelcadi, H., Löw, F., Adams, T, Paulsen, H. (2019). Actinia: cloud based geoprocessing. In Proc. of the 2019 conference on Big Data from Space (BiDS'2019) (pp. 41-44). EUR 29660 EN, Publications Office of the European Union 5, Luxembourg: P. Soille, S. Loekken, and S. Albani (Eds.). (`DOI <https://doi.org/10.5281/zenodo.2631917>`__)
```

(replace with true DOI)

File: doc/quickstart/mapfish_quickstart.rst [hunk 5]
```diff
@@ -93,11 +93,11 @@ At this point you should be able to run the application with::
...
 application that includes a map with two `OSM layers
-<http://www.openstreetmap.org/>`_, a toolbar above the map, and a layer tree.
+<https://www.openstreetmap.org/>`_, a toolbar above the map, and a layer tree.
 
 The MapFish framework offers more than that. In particular it allows easily
 creating RESTful web services for creating, reading, updating and deleting
-features. Please visit http://mapfish.org/doc/2.2/quickstart.html to learn
-how to create such RESTful web services and http://mapfish.org to learn more
+features. Please visit http://www.mapfish.org//doc/2.2/quickstart.html to learn
+how to create such RESTful web services and http://www.mapfish.org/ to learn more
```
Comment (neteler) [inline doc/quickstart/mapfish_quickstart.rst:101]:
```suggestion
features. Please visit http://www.mapfish.org/doc/2.2/quickstart.html to learn
how to create such RESTful web services and http://www.mapfish.org/ to learn more
```

File: doc/quickstart/qgis_quickstart.rst [hunk 6]
```diff
@@ -305,6 +305,6 @@ What next?
 To learn more about QGIS, a good starting point is the `Documentation page`_ on
 the QGIS homepage and `A Gentle Introduction to GIS`_ eBook.
 
-.. _`Documentation page`: https://www.qgis.org/en/docs/index.html
+.. _`Documentation page`: https://www.qgis.org/resources/hub/
 .. _`A Gentle Introduction to GIS`: https://docs.qgis.org/3.4/en/docs/gentle_gis_introduction/index.html
 .. _`QGIS User Guide`: https://docs.qgis.org/3.4/en/docs/user_manual/
```
Comment (neteler) [inline doc/quickstart/qgis_quickstart.rst:310]:
```suggestion
.. _`A Gentle Introduction to GIS`: https://docs.qgis.org/latest/en/docs/gentle_gis_introduction/index.html
.. _`QGIS User Guide`: https://docs.qgis.org/latest/en/docs/user_manual/
```

File: doc/quickstart/vmware_quickstart.rst [hunk 7]
```diff
@@ -29,11 +29,11 @@ Software
 ----------------
 Download the VM version of OSGeoLive. You will need the 7-zip compression program to unzip this file. And of course the VMWare Player software for running the virtual machine. Note that VMWare Player, while offered as a cost-free download, is not open source. You must register on the VMWare website, and agree to their licence in order to use the software.:
 
-`The OSGeo virtual machine  <http://download.osgeo.org/livedvd/release/5.5/osgeolive-vm-5.5.7z>`_
+`The OSGeo virtual machine  <https://download.osgeo.org/livedvd/releases/5.5/>`_
```
Comment (neteler) [inline doc/quickstart/vmware_quickstart.rst:32]: V5.5 is very old, perhaps point to the parent directory (and update the file header above?).

File: locale/en/LC_MESSAGES/quickstart/actinia_quickstart.po [hunk 8]
```diff
@@ -449,7 +450,7 @@ msgid ""
 "based geoprocessing. In Proc. of the 2019 conference on Big Data from "
 "Space (BiDS'2019) (pp. 41-44). EUR 29660 EN, Publications Office of the "
 "European Union 5, Luxembourg: P. Soille, S. Loekken, and S. Albani "
-"(Eds.). (`DOI <https://zenodo.org/record/2631917>`__)"
+"(Eds.). (`DOI <https://zenodo.org/records/2631917>`__)"
```
Comment (neteler) [inline locale/en/LC_MESSAGES/quickstart/actinia_quickstart.po:453]:
```suggestion
"(Eds.). (`DOI <https://doi.org/10.5281/zenodo.2631917>`__)"
```

File: locale/pot/quickstart/actinia_quickstart.pot [hunk 9]
```diff
@@ -344,7 +344,7 @@ msgid "actinia tutorial: `https://neteler.gitlab.io/actinia-introduction <https:
 msgstr ""
 
 #: ../../build/doc/quickstart/actinia_quickstart.rst:482
-msgid "Further reading: Neteler, M., Gebbert, S., Tawalika, C., Bettge, A., Benelcadi, H., Löw, F., Adams, T, Paulsen, H. (2019). Actinia: cloud based geoprocessing. In Proc. of the 2019 conference on Big Data from Space (BiDS'2019) (pp. 41-44). EUR 29660 EN, Publications Office of the European Union 5, Luxembourg: P. Soille, S. Loekken, and S. Albani (Eds.). (`DOI <https://zenodo.org/record/2631917>`__)"
+msgid "Further reading: Neteler, M., Gebbert, S., Tawalika, C., Bettge, A., Benelcadi, H., Löw, F., Adams, T, Paulsen, H. (2019). Actinia: cloud based geoprocessing. In Proc. of the 2019 conference on Big Data from Space (BiDS'2019) (pp. 41-44). EUR 29660 EN, Publications Office of the European Union 5, Luxembourg: P. Soille, S. Loekken, and S. Albani (Eds.). (`DOI <https://zenodo.org/records/2631917>`__)"
```
Comment (neteler) [inline locale/pot/quickstart/actinia_quickstart.pot:347]:
```suggestion
msgid "Further reading: Neteler, M., Gebbert, S., Tawalika, C., Bettge, A., Benelcadi, H., Löw, F., Adams, T, Paulsen, H. (2019). Actinia: cloud based geoprocessing. In Proc. of the 2019 conference on Big Data from Space (BiDS'2019) (pp. 41-44). EUR 29660 EN, Publications Office of the European Union 5, Luxembourg: P. Soille, S. Loekken, and S. Albani (Eds.). (`DOI <https://doi.org/10.5281/zenodo.2631917>`__)"
```

File: doc/quickstart/vmware_quickstart.rst (hunk 7 above)
Comment (geographika) [reply doc/quickstart/vmware_quickstart.rst:32]: @neteler `vmware_quickstart.rst` isn't linked to from anywhere and isn't available at https://live.osgeo.org/en/quickstart/vmware_quickstart.html so I'm thinking this whole file is outdated and could be removed?

File: doc/conf.py [hunk 10]
```diff
@@ -227,19 +232,27 @@
...
 
         # Link to the presentation
-        r'presentation.html'
+        r'presentation.html',
+
+        # Ignore all Stack Overflow links
+        # as they return 429 Client Error: Too Many Requests for url
+        # due to rate limiting
+        r'https://stackoverflow\.com/.*'
```
[the remark this answers is not in the document]
Comment (geographika) [reply doc/conf.py:255]: No I don't think so - I'll remove this.

File: doc/conf.py (hunk 1 above)
Comment (geographika) [reply doc/conf.py:242]: The URLs in the conf.py are just URLs to ignore when reporting errors. I can do a global replace for all `webchat.freenode.net` to `web.libera.chat` - there are 22 instances of these. Maybe the URL is also valid so this line in conf.py can then be removed.

**Comment by neteler** [comment]
(I have fixed the ZOO URLs in 036423623c88caf91f8a80a2c6d91a6a08ddd492, now the CI passes.)

File: doc/conf.py (hunk 2 above)
Comment (geographika) [reply doc/conf.py]: Updated

File: doc/conf.py (hunk 1 above)
Comment (geographika) [reply doc/conf.py]: Updated

File: doc/conf.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (geographika) [reply doc/conf.py]: Updated

**Comment by geographika** [comment]
Thanks for the review @neteler. I've addressed all the comments, apart from the VMWare link. [The VMWare page](https://github.com/OSGeo/OSGeoLive-doc/blob/master/doc/quickstart/vmware_quickstart.rst) is very outdated (referring to OSGeoLive 5.5), so should either be removed or updated and tested in a separate pull request.

**Pull request merged**